modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/TcpDiscoverySpiLogTest.java

```diff
@@ -0,0 +1,225 @@
+
+import static org.apache.ignite.testframework.GridTestUtils.waitForCondition;
+
+/**
+ * Class for {@link TcpDiscoverySpi} logging tests.
+ */
+public class TcpDiscoverySpiLogTest extends GridCommonAbstractTest {
+    /**
+     * Listener log messages
+     */
+    private static ListeningTestLogger testLog;
+
+    /** */
+    private static final ThreadLocal<TcpDiscoverySpi> nodeSpi = new ThreadLocal<>();
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        IgniteConfiguration cfg = super.getConfiguration(igniteInstanceName);
+
+        TcpDiscoverySpi spi = nodeSpi.get();
+
+        if (spi == null)
+            spi = new TcpDiscoverySpi();
+        else
+            nodeSpi.set(null);
+
+        cfg.setDiscoverySpi(spi);
+
+        cfg.setGridLogger(testLog);
+
+        return cfg;
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTest() throws Exception {
+        super.beforeTest();
+
+        testLog = new ListeningTestLogger(log);
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTest() throws Exception {
+        testLog.clearListeners();
+
+        super.afterTest();
+    }
+
+    /**
+     * This test checks all the possible logs from socket connection after coordinator quiet close.
+     * <p>
+     * There are 3 nodes used in the test. After the coordinator closure, the 3rd node will try to check the connection
+     * to the 2nd one. In other words the 3rd will try to connect to all effective addresses of the 2nd through socket,
+     * which can be accessed with 3's Discovery SPI.
+     * <p>
+     * For us to get multiple effective addresses for connection, we override
+     * {@link TcpDiscoverySpi#getEffectiveNodeAddresses(TcpDiscoveryNode)} method in
+     * {@link TestCustomEffectiveNodeAddressesSpi} to return custom collection of {@link InetSocketAddress}.
+     * We only need the custom collection once, when 3rd node is trying to probe the 2nd one. As the method is used in
+     * different other scenarios throughout the test, we restrain the method to deviate from expected behaviour only
+     * for the test case, that is when {@link TcpDiscoveryNode#discoveryPort()} is 47501.
+     * <p>
+     * By manipulating the collection order we can check the failing, successful and ignored connection logs at once.
+     *
+     * @throws Exception If failed.
+     * @see TcpDiscoverySpi#getEffectiveNodeAddresses(TcpDiscoveryNode)
+     * @see TestCustomEffectiveNodeAddressesSpi
+     */
+    @Test
+    public void testMultipleSocketConnectionLogMessage() throws Exception {
+        LogListener lsnr0 = LogListener.matches(s ->
+                s.contains("Connection check to node") && s.contains("result=success"))
+            .build();
+
+        LogListener lsnr1 = LogListener.matches(s ->
+                s.contains("Connection check to node") && s.contains("result=skipped"))
+            .build();
```

Comment:
Let's add expected LogListener#Builder#times(int n) everywhere (in the other places too).